File: mobile/src/main/java/info/papdt/express/helper/ui/items/DetailsTwoLineItem.java
package info.papdt.express.helper.ui.items;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.annotation.StringDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

public class DetailsTwoLineItem {

	private String type, content, optional;

	public static final String TYPE_NAME = "name", TYPE_NUMBER = "number";

	@StringDef({TYPE_NAME, TYPE_NUMBER})
	@Retention(RetentionPolicy.SOURCE)
	public @interface ItemType {}

	public DetailsTwoLineItem(@ItemType String type, @NonNull String content) {
		this.type = type;
		this.content = content;
	}

	public DetailsTwoLineItem(@ItemType String type, @NonNull String content, @Nullable String optional) {
		this.type = type;
		this.content = content;
		this.optional = optional;
	}

	public @ItemType String getType() {
		return this.type;
	}

	public @NonNull String getContent() {
		return this.content;
	}

	public String getOptional() {
		return this.optional;
	}

}

File: mobile/src/main/kotlin/info/papdt/express/helper/ui/items/DetailsTwoLineItem.kt
package info.papdt.express.helper.ui.items

import androidx.annotation.StringDef

class DetailsTwoLineItem {

    @get:ItemType
    var type: String? = null
        private set
    var content: String? = null
        private set
    var optional: String? = null

    @StringDef(TYPE_NAME, TYPE_NUMBER)
    @Retention(AnnotationRetention.SOURCE)
    annotation class ItemType()

    constructor(@ItemType type: String, content: String) {
        this.type = type
        this.content = content
    }

    constructor(@ItemType type: String, content: String, optional: String?) {
        this.type = type
        this.content = content
        this.optional = optional
    }

    companion object {

        const val TYPE_NAME = "name"
        const val TYPE_NUMBER = "number"

    }

}
